context-free.md

```diff
@@ -1,21 +1,24 @@
 # Context Free
 
-##  Who is this programming language for?
-
+## Who is this programming language for?
+for people who want to generate, instead of draw, certain kind of shape through computer.
 
 ## What is easy to do in this language? Why is it easy?
-
+It is easy to generate shapes people want because we do not need to know much function to do that. And the parameters are easy to learn and understand. In addition, the code in general is easy to read and understand. So even you are not familiar with coding, you can still do that.
 
 ## What is hard to do in this language? Why is it hard?
-
+In my opinion, it is hard for me to do something else. Like the variables cannot be modified and the loop statement is not familiar to me.
 
 ## How did you learn how to program in this language?
-
+I read through the documentation about it in github.com and saw some of the example codes online.
 
 ## Describe what happens when a ContextFree program runs
 _Specifically, how does the computer interpret the rules in a program, in
 order to turn those rules into a picture?_
-
+Fisrt, the program will "translate" your code into its host language.(I assume this is a internal DSL because 1. I found it has some similar grammar to C++. 2. I did not see the parser)
+Second, it will generate(render) 2D images followed by your code(rules).
+Third, you can see how your code looks like and modified it to look better.
+Then, you can save the image into png format and All done!
 
 ## What do you think is interesting about the ContextFree program you wrote?
-
+So I think the interesting part is I did not use much word on the code. My code seems simple and easy to read. Also, I do not need to define some variables before using them and I do not need to import anything. Plus, no need for mathematical or logic operator. Just your shape and some parameters.
```

Comment:
What do you think may be some positives and negatives on the compactness of this programming language?